File: 0153-find-minimum-in-rotated-sorted-array/0153-find-minimum-in-rotated-sorted-array.cpp
class Solution {
public:
    int findMin(vector<int>& nums) {
        int minm = INT_MAX;
        int n = nums.size();
        for(int i = 0; i<= n; i++){
           minm = min(minm, nums[i%n]);
        }
        return minm;
    }
};

File: 0153-find-minimum-in-rotated-sorted-array/0153-find-minimum-in-rotated-sorted-array.java
class Solution {
    public int findMin(int[] arr) {
        int n = arr.length;

        int l = 0, h = n-1;
        if(arr[l] < arr[h]){
            return arr[l];
        }

        while(l < h){
            int mid = (l+h)/2;
             if(arr[mid] > arr[h]){
                l = mid+1;
            }else{
                h = mid;
            }
        }
        return arr[h];
    }
}